python/cuml/dask/ensemble/base.py

```diff
@@ -85,6 +86,20 @@ def _estimators_per_worker(self, n_estimators):
     def _fit(self, model, dataset, convert_dtype):
         data = DistributedDataHandler.create(dataset, client=self.client)
         self.datatype = data.datatype
+        if self.datatype == 'cudf':
+            dtype = dataset[0].dtypes
+            check = (dtype.any() == np.float64)
+            self.num_classes = len(dataset[1].unique())
+
+        else:
+            dtype = dataset[0].dtype
+            check = (dtype == np.float64)
+            self.num_classes = \

```

Comment:
also, you might want to just raise a typeerror right away when computing check so that we don't do extra work in dask when we're going to fail

Reply:
Yes, we do check the num_classes for the child models but i thought it would be better to check the num_classes when we have the whole dataset. That way we get the right number of classes irrespective of how the dataset gets split up into different chunks/ partitions.

Reply:
In addition, when num_classes>2 we do not support pickling. Therefore, reading the num_classes from the models created in each worker will not be possible.
Furthermore, the CPU implementation of predict for dask RF classification is different from the CPU implementation used for predict in single gpu RF.